yare-core/src/main/java/com/sabre/oss/yare/core/EngineListener.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * MIT License
+ *
+ * Copyright 2018 Sabre GLBL Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.yare.core;
+
+public interface EngineListener {
```

Comment:
I am afraid that I have been misunderstood. I think there should be separate listener for every functionality. I would remove `EngineListener`, `EngineControllerEvent` and `EngineControllerEventType`. I would simply create `StopProcessingListener` which would be an interface with one method `onStopProcessing(StopProcessingEvent)` where `StopProcessingEvent` would be a empty interface right now.

The fact that it would be implemented like this would be convention for next functionality in `EngineController`. Any new method(s) added as a feature would get appropriate listener. There would be only one implementation of `EngineController` which allows for listener registration and which would be reused in both: yare implementation and client tests.